server/internal/core/application/covenantless_test.go

```diff
@@ -0,0 +1,130 @@
+package application
+
+import (
+	"testing"
+	"time"
+)
+
+func TestNextMarketHour(t *testing.T) {
+	marketHourStartTime := parseTime("2023-10-10 13:00:00") //13 16 19 22 01 04
+	marketHourEndTime := parseTime("2023-10-10 13:05:00")
+	period := int64(3 * 3600) // 3 hours in seconds
+
+	testCases := []struct {
+		now           time.Time
+		expectedStart time.Time
+		expectedEnd   time.Time
+		expectError   bool
+		description   string
+	}{
+		{
+			now:           parseTime("2023-10-10 13:00:00"),
+			expectedStart: parseTime("2023-10-10 13:00:00"),
+			expectedEnd:   parseTime("2023-10-10 13:05:00"),
+			expectError:   false,
+			description:   "Now is exactly at the initial market hour start time",
+		},
+		{
+			now:           parseTime("2023-10-10 13:02:00"),
+			expectedStart: parseTime("2023-10-10 16:00:00"),
+			expectedEnd:   parseTime("2023-10-10 16:05:00"),
```

Comment:
in this case we should return the current market hour not the next one

Reply:
Are we sure about this? if user invokes GetInfo in the middle of MarketHour isnt it better to return next market hour?